### PR TITLE
Add und-only baked testdata to repo and make `cargo make quick` quick again

### PR DESCRIPTION
**Pull request by sffc**
With some basic instructions on how to use it with VSCode.

I want to make a cargo alias `cargo quicky` that sets this environment variable and runs `cargo clippy` but I haven't figured out yet how to do that; cargo aliases don't seem to have a lot of flexibility.

### CODE REVIEW

File: provider/datagen/src/tests/make_testdata.rs [hunk 1]
```diff
@@ -55,11 +65,63 @@ fn generate_json_and_verify_postcard() {
...
+        // do not put any payloads in stubdata!
+        Ok(())
+    }
+
+    fn flush_singleton(
+        &self,
+        key: DataKey,
+        payload: &DataPayload<ExportMarker>,
+    ) -> Result<(), DataError> {
+        self.0.flush_singleton(key, payload)
```
Comment (sffc) [inline provider/datagen/src/tests/make_testdata.rs:101]: Note that the singleton keys are still being exported. If I don't export them or if I export them as non-singleton, I get errors like

```
no associated item named `SINGLETON_LOCID_TRANSFORM_ALIASES_V1` found for struct `Baked` in the current scope
associated item not found in `Baked`
```

I'm not sure how to work around this other than to require that all data structs implement `Default` and then plumb through a function that resets the payload to `Default`.

**Comment by sffc** [comment]
Another option is I just export all of the testdata locales into stubdata; it will still be a lot faster than full testdata, and then at least _some_ tests will continue to work.

**Comment by Manishearth** [comment]
Just do the alias with cargo make.

**Comment by sffc** [comment]
> Just do the alias with cargo make.

I want to be able to use the alias from anywhere in tree, but I think cargo make only works if your current working directory is the top level.

**Comment by Manishearth** [comment]
In that case I'd recommend just writing a chmod'd shell script cargo-quicky in your PATH

(I don't think most of us use the cargo aliases anyway, we don't necessarily need to solve this problem for anyone. I'm happy with a make target that accepts a folder)

File: provider/datagen/src/tests/make_testdata.rs [hunk 2]
```diff
@@ -46,6 +47,15 @@ fn generate_json_and_verify_postcard() {
         ),
     });
 
+    let stubdata_out = Box::new(BakedStubdataExporter(
+        baked_exporter::BakedExporter::new("tests/data/stub".into(), {
```
Comment (robertbastian) [inline provider/datagen/src/tests/make_testdata.rs:51]: This is not only useful for build times, but also for debugging baked output. With that in mind, please
* change the path to `tests/data/baked`
* set `pretty`
* move the VSCode instructions to `CONTRIBUTING.md` or somewhere else. They're not relevant for this test
* consider not filtering the `und` payload if size allows it. We're already including singletons anyway.

**Comment by robertbastian** [comment]
> (I don't think most of us use the cargo aliases anyway, we don't necessarily need to solve this problem for anyone. I'm happy with a make target that accepts a folder)

Yeah I feel like cargo aliases should not be a workspace thing, everyone can just have their own

**Comment by sffc** [comment]
Autosubmit = true

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -20,6 +22,26 @@ To build ICU4X, you will need the following dependencies:
...
+
+ICU4X can be edited using any text editor capable of editing Rust code.
+
+#### Visual Studio Code
+
+Many ICU4X engineers use [Visual Studio Code](https://code.visualstudio.com/) with the [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer) extension.
+
+To build all code paths and improve build times in VSCode, we recommend the following settings. (Note: The second setting causes VSCode to build ICU4X with only the `und` locale, which reduces build times but also makes some tests fail; to run them normally, run `cargo test --all-features` on the command line.) To add these settings, choose "Preferences: Open Workspace Settings (JSON)" from the command palette (Ctrl+Shift+P):
+
+```javascript
```
Comment (robertbastian) [inline CONTRIBUTING.md:35]: this isn't formatting correctly, you probably need
```suggestion
```json
```

File: Makefile.toml [hunk 4]
```diff
@@ -26,13 +26,12 @@ ICU4X_BUILDING_WITH_FORCED_NIGHTLY = { value = "1", condition = { env_set = ["IC
 [tasks.quick]
 description = "Run quick version of all lints and builds (useful before pushing to GitHub)"
 category = "ICU4X Development"
-dependencies = [
+env = { "ICU4X_DATA_DIR" = "${CARGO_MAKE_WORKSPACE_WORKING_DIRECTORY}/provider/datagen/tests/data/baked" }
+run_task.name = [
     "ci-job-fmt",
     "ci-job-clippy",
-    "ci-job-nostd",
```
Comment (robertbastian) [inline Makefile.toml:32]: nit: reorder the jobs in `ci-job-all` as well

File: provider/datagen/src/tests/make_testdata.rs (hunk 1 above)
Comment (robertbastian) [reply provider/datagen/src/tests/make_testdata.rs:108]: This is kind of the point of singleton keys, the data structs has to be present. 

I think implementing `Default` would be very bad, as that would be part of the public API. We don't want anyone to construct a data struct like that. You'd also have to pipe this through the `dyn Exportable`, which means either all structs have to implement `Default`, or you need custom filtering code in `icu_provider`. I think this is a no-go.

You can implement custom filtering here by matching on keys:
```rust
self.0.flush_singleton(
  key,
  match key {
    LineBreakDataV1Marker:KEY => &DataPayload::from_owned(...).upcast(),
    // ...
    _ => payload
  } 
}
```

File: .github/workflows/build-test.yml [hunk 5]
```diff
@@ -288,6 +288,8 @@ jobs:
       with:
         key: download-cache
         path: /tmp/icu4x-source-cache
+    - name: Install rustfmt
```
Comment (robertbastian) [inline .github/workflows/build-test.yml:291]: Can you test how much time formatting costs?

File: provider/datagen/src/tests/make_testdata.rs (hunk 1 above)
Comment (sffc) [reply provider/datagen/src/tests/make_testdata.rs:108]: Now that we're including `und` for all keys it is no longer inconsistent to include it for singleton keys so I consider this thread to be obsolete.

File: CONTRIBUTING.md (hunk 3 above)
Comment (sffc) [reply CONTRIBUTING.md:35]: I can't have comments in JSON; they appear red highlighted. I don't see what's wrong with the JavaScript formatting.

File: Makefile.toml (hunk 4 above)
Comment (sffc) [reply Makefile.toml:32]: I did not change the order; I removed some things from the job entirely. I figured it was fairly rare for the nostd job to randomly fail, so I just removed it from `cargo make quick`. Same with `tutorials-local` and certain parts of `tidy`. Everything still runs on CI; this is just to catch the things that commonly don't work, like Clippy, Fmt, and Docs.

File: .github/workflows/build-test.yml (hunk 5 above)
Comment (sffc) [reply .github/workflows/build-test.yml:291]: With fmt:

```
real	0m18.767s
user	1m2.729s
sys	0m4.626s

real	0m19.310s
user	1m21.325s
sys	0m5.306s

real	0m19.624s
user	1m22.069s
sys	0m5.376s
```

Without fmt:

```
real	0m16.967s
user	1m12.534s
sys	0m0.914s

real	0m12.848s
user	0m48.084s
sys	0m0.730s

real	0m16.216s
user	0m57.994s
sys	0m0.824s
```

**Comment by sffc** [comment]
Still autosubmit = true

File: CONTRIBUTING.md (hunk 3 above)
Comment (robertbastian) [reply CONTRIBUTING.md:35]: Well it's all just blue, JSON formatting highlights keys and values: https://github.com/unicode-org/icu4x/blob/7a7e93cf8790968970ecef7adb1610958f485e81/ffi/npm/package.json#L4

File: Makefile.toml (hunk 4 above)
Comment (robertbastian) [reply Makefile.toml:32]: Yes, and the first section in `ci-job-all` that says `# make quick` still contains these. You should move them after that section if you have determined that they are too slow to run in `quick`.

File: provider/datagen/src/tests/make_testdata.rs (hunk 1 above)
Comment (robertbastian) [reply provider/datagen/src/tests/make_testdata.rs:108]: How is the build time? We do have some fairly big `und`s and singletons

File: .github/workflows/build-test.yml (hunk 5 above)
Comment (robertbastian) [reply .github/workflows/build-test.yml:291]: cool thanks.

**Review by robertbastian — APPROVED** [review]

File: provider/datagen/src/tests/make_testdata.rs (hunk 1 above)
Comment (sffc) [reply provider/datagen/src/tests/make_testdata.rs:108]: The build time is significantly smaller than it had been before so I'm happy. The slowest crate to build before was icu_datetime, which has a lot of small data, and icu_datetime is a lot faster with this change.

File: Makefile.toml (hunk 4 above)
Comment (sffc) [reply Makefile.toml:32]: #4559

File: CONTRIBUTING.md (hunk 3 above)
Comment (sffc) [reply CONTRIBUTING.md:35]: Let's see how it looks in #4560

**Pull request merged**